pkg/registry/apis/iam/team/validate.go

```diff
@@ -30,3 +30,28 @@ func ValidateOnCreate(ctx context.Context, obj *iamv0alpha1.Team) error {
 
 	return nil
 }
+
+func ValidateOnUpdate(ctx context.Context, obj, old *iamv0alpha1.Team) error {
+	requester, err := identity.GetRequester(ctx)
+	if err != nil {
+		return apierrors.NewUnauthorized("no identity found")
+	}
+
+	if obj.Spec.Title == "" {
+		return apierrors.NewBadRequest("the team must have a title")
+	}
+
+	if !requester.IsIdentityType(types.TypeServiceAccount) && obj.Spec.Provisioned && !old.Spec.Provisioned {
```

Comment:
It's fine for this now, but probably this should be changed to `TypeAccessPolicy` meaning only internal services (SCIM) are able to create provisioned teams.